external-plugins/rehearse/plugin/server/eventsserver.go

```diff
@@ -0,0 +1,37 @@
+package server
+
+import (
+	"net/http"
+
+	"k8s.io/test-infra/prow/github"
+	"kubevirt.io/project-infra/external-plugins/rehearse/handler"
+)
+
+type GitHubEventsServer struct {
+	tokenGenerator func() []byte
+	eventsHandler  *handler.GitHubEventsHandler
+}
+
+func NewGitHubEventsServer(tokenGenerator func() []byte, eventsChan *handler.GitHubEventsHandler) *GitHubEventsServer {
+	return &GitHubEventsServer{
+		tokenGenerator: tokenGenerator,
+		eventsHandler:  eventsChan,
+	}
+}
+
+func (s *GitHubEventsServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	eventType, eventGUID, eventPayload, eventOk, _ := github.ValidateWebhook(writer, request, s.tokenGenerator)
+
+	if !eventOk {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	event := &handler.GitHubEvent{
+		Type:    eventType,
+		GUID:    eventGUID,
+		Payload: eventPayload,
+	}
+	go s.eventsHandler.Handle(event)
```

Comment:
If we use error propagating to facilitate unit testing we can log the error inside the goroutine so it's just at one place

```golang
go func() {
   err := s.eventsHandler.Handle(event)
   log.WithError(err)....
}
```